packages/toolpad-core/src/DashboardLayout/DashboardLayout.tsx

```diff
@@ -460,6 +465,11 @@ function DashboardLayout(props: DashboardLayoutProps) {
 
   const getMenuIcon = React.useCallback(
     (isExpanded: boolean) => {
+
+      if (hideNavigation) {
```

Comment:
There's also a `marginRight` on the boxes around these icons that we should not have if they're not shown.
So instead on line 566 we can probably have something like:

```tsx
{!hideNavigation ? 
         <Box
            sx={{
              mr: { sm: disableCollapsibleSidebar ? 0 : 1 },
              display: { md: 'none' },
            }}
          >
            {getMenuIcon(isMobileNavigationExpanded)}
          </Box>
          <Box
            sx={{
              display: { xs: 'none', md: disableCollapsibleSidebar ? 'none' : 'block' },
              mr: disableCollapsibleSidebar ? 0 : 1,
            }}
          >
            {getMenuIcon(isDesktopNavigationExpanded)}
          </Box> : null}
```